languages/rust/exercises/concept/options/tests/options.rs

```diff
@@ -2,85 +2,88 @@ use options::*;
+        .expect("reviving a dead player must return Some(player)");
     assert_eq!(revived_player.health, 100);
     assert_eq!(revived_player.mana, Some(100));
-    assert_eq!(revived_player.level, level);
+    assert_eq!(revived_player.level, dead_player.level);
 }
 
 #[test]
 #[ignore]
 fn test_reviving_alive_player() {
-    let level: u32 = 8;
     let alive_player = Player {
         health: 1,
         mana: None,
-        level,
+        level: 8,
     };
-    assert_eq!(alive_player.revive().is_none(), true);
+    assert_eq!(alive_player.revive(), None);
 }
 
 #[test]
 #[ignore]
 fn test_cast_spell_with_enough_mana() {
-    let health = 99;
-    let mana = 100;
-    let level = 100;
-    let mana_cost = 3;
+    const HEALTH: u32 = 99;
+    const MANA: u32 = 100;
+    const LEVEL: u32 = 100;
+    const MANA_COST: u32 = 3;
 
     let mut accomplished_wizard = Player {
-        health,
-        mana: Some(mana),
-        level,
+        health: HEALTH,
+        mana: Some(MANA),
+        level: LEVEL,
     };
 
-    assert_eq!(accomplished_wizard.cast_spell(mana_cost), mana_cost * 2);
-    assert_eq!(accomplished_wizard.health, health);
-    assert_eq!(accomplished_wizard.mana, Some(mana - mana_cost));
-    assert_eq!(accomplished_wizard.level, level);
+    assert_eq!(accomplished_wizard.cast_spell(MANA_COST), MANA_COST * 2);
+    assert_eq!(accomplished_wizard.health, HEALTH);
+    assert_eq!(accomplished_wizard.mana, Some(MANA - MANA_COST));
+    assert_eq!(accomplished_wizard.level, LEVEL);
 }
 
 #[test]
 #[ignore]
 fn test_cast_spell_with_insufficient_mana() {
-    let health = 56;
-    let mana = 0;
-    let level = 22;
-    let mana_cost = 3;
-
     let mut no_mana_wizard = Player {
-        health,
-        mana: Some(mana),
-        level,
+        health: 56,
+        mana: Some(2),
+        level: 22,
     };
 
-    assert_eq!(no_mana_wizard.cast_spell(mana_cost), 0);
-    assert_eq!(no_mana_wizard.health, health);
-    assert_eq!(no_mana_wizard.mana, Some(mana));
-    assert_eq!(no_mana_wizard.level, level);
+    // we want to clone so we can compare before-and-after effects of casting the spell,
+    // but we don't want to introduce that concept to the student yet, so we have to do it manually
+    let clone = Player { ..no_mana_wizard };
+
+    assert_eq!(no_mana_wizard.cast_spell(3), 0);
+    assert_eq!(no_mana_wizard.health, clone.health);
+    assert_eq!(no_mana_wizard.mana, clone.mana);
+    assert_eq!(no_mana_wizard.level, clone.level);
 }
 
 #[test]
 #[ignore]
 fn test_cast_spell_with_no_mana_pool() {
-    let health = 87;
-    let level = 6;
-    let mana_cost = 3;
+    const MANA_COST: u32 = 10;
 
     let mut underleveled_player = Player {
-        health,
+        health: 87,
         mana: None,
-        level,
+        level: 6,
+    };
+
+    let clone = Player {
+        health: underleveled_player.health,
+        mana: underleveled_player.mana,
+        level: underleveled_player.level,
     };
```

Comment:
```suggestion
    let clone = Player { ..underleveled_player };
```

I think that's right... does this repo check compilation

Reply:
I don't think it checks compilation, no. Still, that's how I'd write it, so I'm going to commit it.